xbmc/network/NetworkServices.cpp

```diff
@@ -854,6 +853,42 @@ bool CNetworkServices::StopUPnPClient()
   return false;
 }
 
+bool CNetworkServices::StartUPnPController()
+{
+#ifdef HAS_UPNP
+  if (!CSettings::Get().GetBool("services.upnpcontroller") ||
+      !CSettings::Get().GetBool("services.upnpserver"))
```

Comment:
Not entirely sure if intended or just c/p. Checking for the controller part only won't work, right?